src/http.zig

```diff
@@ -379,15 +379,57 @@ fn NewHTTPContext(comptime ssl: bool) type {
 
         pub fn init(this: *@This()) !void {
             if (comptime ssl) {
-                const opts: uws.us_bun_socket_context_options_t = .{
+                var opts: uws.us_bun_socket_context_options_t = .{
                     // we request the cert so we load root certs and can verify it
                     .request_cert = 1,
                     // we manually abort the connection if the hostname doesn't match
                     .reject_unauthorized = 0,
                 };
+
+                var extra_cas: ?std.ArrayList([*c]const u8) = null;
+
+                blk: {
+                    if (std.process.getEnvVarOwned(default_allocator, "NODE_EXTRA_CA_CERTS")) |extra_ca_certs_path| {
+                        const file = bun.openFile(extra_ca_certs_path, .{ .mode = .read_only }) catch |err| {
```

Comment:
There are way too many ways to read files in Bun's codebase which is very confusing

Instead of this, let's do:
```zig
const contents = bun.sys.File.readFromUserInput(std.fs.cwd(), extra_ca_certs_path, bun.default_allocator)
```

And not in here, instead in `DotEnv.Loader`